app/src/test/java/org/simple/clinic/facility/change/FacilityChangeScreenControllerTest.kt

```diff
@@ -273,4 +273,65 @@ class FacilityChangeScreenControllerTest {
   fun `params for permission denials`(): List<RuntimePermissionResult> {
     return listOf(DENIED, NEVER_ASK_AGAIN)
   }
+
+  @Test
+  @Parameters("6", "7", "8")
+  fun `when location is being fetched then it should expire after a fixed time duration`(
+      secondsSpentWaitingForLocation: Long
+  ) {
+    configProvider.onNext(configTemplate.copy(locationListenerExpiry = Duration.ofSeconds(5)))
+
+    whenever(facilityRepository.facilitiesInCurrentGroup(any(), any())).thenReturn(Observable.just(emptyList()))
+    whenever(locationRepository.streamUserLocation(any(), any())).thenReturn(Observable.never())
+
+    uiEvents.onNext(ScreenCreated())
+    uiEvents.onNext(FacilityChangeLocationPermissionChanged(GRANTED))
+    verify(screen).showProgressIndicator()
+
+    testComputationScheduler.advanceTimeBy(secondsSpentWaitingForLocation, SECONDS)
+    verify(screen).hideProgressIndicator()
+  }
+
+  @Test
+  fun `when screen starts and location permission is available then progress indicator should be shown`() {
+    whenever(locationRepository.streamUserLocation(any(), any())).thenReturn(Observable.never())
+
+    uiEvents.onNext(ScreenCreated())
+    uiEvents.onNext(FacilityChangeLocationPermissionChanged(GRANTED))
+    uiEvents.onNext(FacilityChangeLocationPermissionChanged(GRANTED))
+
+    verify(screen, times(1)).showProgressIndicator()
+  }
+
+  @Test
+  @Parameters(method = "params for permission denials")
+  fun `when screen starts and location permission is unavailable then progress indicator should not be shown`(
```

Comment:
This test checks if location permission was `denied` and not `unavailable` whereas the test name says "unavailable".
Can you check and correct this?

Reply:
I wanted to refer to `denied` and `rejected permanently` with one word. `unavailable` is what came to my mind. Do you have a better suggestion?

Reply:
Both `DENIED` and `NEVER ASK AGAIN` point to permission being denied.
`Unavailable` sounds like misleading in this context.

Reply:
Renamed `is unavailable` -> `was denied`.